tchannel/errors.py

```diff
@@ -35,97 +42,107 @@ class ProtocolError(TChannelError):
         'tracing',
     )
 
+    code = None
+
     def __init__(
         self,
-        code,
-        description,
+        description=None,
         id=None,
         tracing=None,
     ):
         super(TChannelError, self).__init__(description)
-        self.code = code
         self.tracing = tracing
         self.id = id
         self.description = description
 
+    @classmethod
+    def from_code(cls, code, **kw):
+        return {
+            TIMEOUT: TimeoutError,
+            CANCELED: CanceledError,
+            BUSY: BusyError,
+            DECLINED: DeclinedError,
+            UNEXPECTED_ERROR: UnexpectedError,
+            BAD_REQUEST: BadRequestError,
+            NETWORK_ERROR: NetworkError,
+            UNHEALTHY: UnhealthyError,
+            FATAL: FatalProtocolError,
+        }[code](**kw)
+
+
+class AlwaysRetryableError(TChannelError):
+    pass
+
 
-class InvalidMessageError(TChannelError):
-    """Represent an invalid message."""
+class PossiblyRetryableError(TChannelError):
```

Comment:
I agree that 'always retryable' and 'possibly retryable' could be confusing; especially because always retryable sounds like immediately and infinitely retryable. 

How about just Retryable and with some sort of property that is true if the original request might have made it to the server?